PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightsJavaProjectWizard.java

```diff
@@ -115,15 +142,17 @@ public boolean canFlipToNextPage() {
 			Display.getDefault().syncExec(new Runnable() {
 				@Override
 				public void run() {
-					final String jarPathString = sparkLibraryOptionsPanel.getSparkLibraryPath();
-					if (StringUtils.isNullOrEmpty(jarPathString)) {
-						DefaultLoader.getUIHelper().showError("Spark Library Path cannot be null",
-								"Spark Project Settings");
-					} else {
-						IPath jarPath = new Path(jarPathString);
-						IClasspathEntry sparkEntry = JavaCore.newLibraryEntry(jarPath, null, null);
-						System.arraycopy(entries, 0, newEntries, 0, entries.length);
-						newEntries[entries.length] = sparkEntry;
+					if (!sparkLibraryOptionsPanel.useMaven) {
+						final String jarPathString = sparkLibraryOptionsPanel.getSparkLibraryPath();
+						if (StringUtils.isNullOrEmpty(jarPathString)) {
+							DefaultLoader.getUIHelper().showError("Spark Library Path cannot be null",
+									"Spark Project Settings");
+						} else {
+							IPath jarPath = new Path(jarPathString);
+							IClasspathEntry sparkEntry = JavaCore.newLibraryEntry(jarPath, null, null);
+							System.arraycopy(entries, 0, newEntries, 0, entries.length);
+							newEntries[entries.length] = sparkEntry;
```

Comment:
Will it be out of array bound?

Reply:
Rewrote this part

---
In reply to: [136774700](https://github.com/Microsoft/azure-tools-for-java/pull/984#discussion_r136774700) [](ancestors = 136774700)

Reply:
Checked, it will not

---
In reply to: [138295140](https://github.com/Microsoft/azure-tools-for-java/pull/984#discussion_r138295140) [](ancestors = 138295140,136774700)